src/IlkRegistry.sol

```diff
@@ -85,29 +96,33 @@ contract IlkRegistry {
     GemInfo  private immutable gemInfo;
 
     CatLike  public cat;
+    DogLike  public dog;
     SpotLike public spot;
 
     struct Ilk {
-        uint256 pos;   // Index in ilks array
-        address gem;   // The token contract
-        address pip;   // Token price
-        address join;  // DSS GemJoin adapter
-        address flip;  // Auction contract
-        uint256 dec;   // Token decimals
-        string name;   // Token name
-        string symbol; // Token symbol
+        uint96  pos;    // Index in ilks array
+        address join;   // DSS GemJoin adapter
+        address gem;    // The token contract
+        uint8   dec;    // Token decimals
+        uint96  class;  // Classification code (1 - clip, 2 - flip, 3+ - other)
+        address pip;    // Token price
+        address xlip;   // Auction contract
+        string  name;   // Token name
+        string  symbol; // Token symbol
     }
 
     mapping (bytes32 => Ilk) public ilkData;
     bytes32[] ilks;
 
     // Initialize the registry
-    constructor(address vat_, address cat_, address spot_) public {
+    constructor(address vat_, address dog_, address cat_, address spot_) public {
```

Comment:
Super super nit: we might want to keep the `cat` - `dog` order everywhere.

Reply:
Yeah, some things got moved around to optimize for clipper collaterals. I'll make the update.